### ISSUE TITLE
Update for gnome 3.36 ?

### ISSUE THREAD

**Issue by galen1423** [rostegg/public-ip-gnome-extension#15]
Thanx for this usefull extension.

I use Ubuntu 20.04 with GS 3.35.91 and public-ip return the following error

`Error: Tried to construct an object without a GType; are you using GObject.registerClass() when inheriting from a GObject type?`

I hope that you will maintained code with GS 3.36...

**Comment by rostegg** [comment]
Hey. at the moment I can’t reproduce the bug (because I can’t deploy the testing environment, because now there are only daily images and they are not very stable deployed on virtual machines), so that as soon as I can fix the error and solve it, I'll let you know

**Comment by galen1423** [comment]
Hi rostegg,

No problem, i will waiting quietly for updates... :-)

Thanx.

**Comment by aminvakil** [comment]
Arch Linux has been updated to gnome-shell 3.35 which is beta for 3.36 and it will update to gnome 3.36 tomorrow on March 11.

I'm just putting this here and anyone can read this pull request which fixes the issue with gnome 3.36 and fix issue in this repo as well:
https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/pull/564

**Comment by rostegg** [comment]
Okay, i created a hotfix [here](https://github.com/rostegg/public-ip-gnome-extension/tree/gnome-3.36-hotfix)([commit](https://github.com/rostegg/public-ip-gnome-extension/commit/88a5788dedb207f09cdec0917ddee464c4dfc7b4)).  

It's work ok on Ubuntu 20.04, GS 3.35.91 (yes, I finally managed to run it).

Now I will not make changes in master branch, because I want to understand [what they did and, most importantly, why](https://gitlab.gnome.org/GNOME/gjs/commit/72062b5e036821731116a9263b55ee647e3bdadf).

In a couple of days, when I make sure that it works as it should and is compatible with previous versions, I will make changes to the master branch.

**Comment by aminvakil** [comment]
In case you haven't noticed it's working ok but if you run `sudo journalctl -f` and restart gnome-shell using Alt + F2 and r, you will get this warning which I assume it's OK and you're willing to fix it stable release.
```
Usage of object.actor is deprecated for IpInfoIndicator
get@resource:///org/gnome/shell/ui/environment.js:285:29
_init@/usr/share/gnome-shell/extensions/public-ip-extension@rostegg.github.com/extension.js:192:5
enable@/usr/share/gnome-shell/extensions/public-ip-extension@rostegg.github.com/extension.js:274:35
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:160:32
loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:311:26
_loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:536:18
collectFromDatadirs@resource:///org/gnome/shell/misc/fileUtils.js:27:17
_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:515:19
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:545:18
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:576:18
init@resource:///org/gnome/shell/ui/extensionSystem.js:54:14
_initializeUI@resource:///org/gnome/shell/ui/main.js:244:22
start@resource:///org/gnome/shell/ui/main.js:138:5
@<main>:1:47
```

Also if you want to go to settings from gnome-tweaks, it gives me this error:

```
TypeError: widget.show_all is not a function

Stack trace:
  buildPrefsWidget@/usr/share/gnome-shell/extensions/public-ip-extension@rostegg.github.com/prefs.js:106:9
  _showPrefs@resource:///org/gnome/shell/extensionPrefs/main.js:174:38
  _extensionsLoaded@resource:///org/gnome/shell/extensionPrefs/main.js:467:18
  _scanExtensions/<@resource:///org/gnome/shell/extensionPrefs/main.js:418:18
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
  main@resource:///org/gnome/shell/extensionPrefs/main.js:762:23
  @<main>:1:48
```

**Comment by rostegg** [comment]
> 
> 
> In case you haven't noticed it's working ok but if you run `sudo journalctl -f` and restart gnome-shell using Alt + F2 and r, you will get this warning which I assume it's OK and you're willing to fix it stable release.
> 
> ```
> Usage of object.actor is deprecated for IpInfoIndicator
> get@resource:///org/gnome/shell/ui/environment.js:285:29
> _init@/usr/share/gnome-shell/extensions/public-ip-extension@rostegg.github.com/extension.js:192:5
> enable@/usr/share/gnome-shell/extensions/public-ip-extension@rostegg.github.com/extension.js:274:35
> _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:160:32
> loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:311:26
> _loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:536:18
> collectFromDatadirs@resource:///org/gnome/shell/misc/fileUtils.js:27:17
> _loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:515:19
> _enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:545:18
> _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:576:18
> init@resource:///org/gnome/shell/ui/extensionSystem.js:54:14
> _initializeUI@resource:///org/gnome/shell/ui/main.js:244:22
> start@resource:///org/gnome/shell/ui/main.js:138:5
> @<main>:1:47
> ```

Fixed with [this](https://github.com/rostegg/public-ip-gnome-extension/commit/75bdac619820478acd53e9aa9c2be4d203395e23)



> Also if you want to go to settings from gnome-tweaks, it gives me this error:
> 
> ```
> TypeError: widget.show_all is not a function
> 
> Stack trace:
>   buildPrefsWidget@/usr/share/gnome-shell/extensions/public-ip-extension@rostegg.github.com/prefs.js:106:9
>   _showPrefs@resource:///org/gnome/shell/extensionPrefs/main.js:174:38
>   _extensionsLoaded@resource:///org/gnome/shell/extensionPrefs/main.js:467:18
>   _scanExtensions/<@resource:///org/gnome/shell/extensionPrefs/main.js:418:18
>   asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
>   main@resource:///org/gnome/shell/extensionPrefs/main.js:762:23
>   @<main>:1:48
> ```

Well, I'm not sure about [this fix](https://github.com/rostegg/public-ip-gnome-extension/commit/9482f00877b4c14bfa674ffab7ffc456022ff493), because all my test envs work weid, so I can't open settings with tweaks, but it's not give me any error. So, I will be grateful if someone checks to see if the fix works, and if not, he will report an error and a stack trace.

Also, for a while, there will be a separate branch for version 3.36, because these corrections no longer work in version 3.34 and lower. After reading what the rest of the developers write, I have not yet found a more or less normal solution.

**Comment by aminvakil** [comment]
Both errors are fixed and as far as I can see nothing is wrong (settings works as well)

**Comment by rostegg** [comment]
For now, extension for version 3.36 will be in a [separate branch](https://github.com/rostegg/public-ip-gnome-extension/tree/gnome-3.36-hotfix).

I've already notified everywhere about the need for manual installation for version 3.36.

I noticed that almost all extension began to have problems when switching to version 3.36. So that I will follow how this problem will be solved and as soon as there is a good solution for compatibility with earlier versions of gnome shell, I will definitely refactor code.
